src/main/scala/org/renci/robocord/RoboCORD.scala

```diff
@@ -82,37 +105,21 @@ object RoboCORD extends App with LazyLogging {
   val articlesTotal = metadata.size
   logger.info(s"Selected $articlesTotal articles for processing (from $startIndex until $endIndex, chunk $currentChunk out of $totalChunks)")
 
-  // Identify full text SHAs and PMCIDs in the current chunk (`metadata`).
-  val shasToLoad = metadata.flatMap(_.get("sha")).flatMap(_.split(';')).map(_.trim.toLowerCase).toSet
-  val pmcIdsToLoad = metadata.flatMap(_.get("pmcid")).flatMap(_.split(';')).map(_.trim.toLowerCase).toSet
-  // logger.info(s"shasToLoad: $shasToLoad")
-  // logger.info(s"pmcIdsToLoad: $pmcIdsToLoad")
-
-  // Load up full text data for those SHAs and PMCIDs, and create a map so we can identify them easily.
-  val fullTextData: Seq[CORDArticleWrapper] = conf.data().flatMap(CORDJsonReader.wrapFileOrDir(_, logger, shasToLoad, pmcIdsToLoad))
-  val fullTextById: Map[String, Seq[CORDArticleWrapper]] = fullTextData.groupBy(_.id)
-  logger.info(s"Loaded ${fullTextData.size} full text documents corresponding to ${fullTextById.keySet.size} IDs.")
-
   // Run SciGraph in parallel over the chunk we need to process.
   logger.info(s"Starting SciGraph in parallel on ${Runtime.getRuntime.availableProcessors} processors.")
 
   var articlesCompleted = 0
 
   // Summarize all files into the output directory.
-  val results: ParSeq[String] = metadata.par.flatMap(entry => {
+  val results: ParSeq[String] = metadata.zipWithIndex.par.flatMap({ case (entry, index) =>
     // Find the ID of this article.
-    val id = entry("cord_uid")
+    // We have some duplicate CORD_UIDs (yes, really!). So we add the metadata row index to make it fully unique.
+    val id = s"${entry("cord_uid")}:${startIndex + index}"
     val pmid = entry.get("pubmed_id")
     val doi = entry.get("doi")
 
-    // For SHA and PMCID, we do a little additional processing: if there are multiple SHAs or PMCIDs,
-    // we only use the last one, but we record all IDs in the log.
-    val shas = entry.getOrElse("sha", "").split(';').map(_.trim).filter(_.nonEmpty)
-    val sha = if (shas.length > 1) {
-      val shaLast = shas.last
-      logger.info(s"Found multiple SHAs for $id, choosing the last one ($shaLast) out of: $shas")
-      shaLast
-    } else shas.headOption.getOrElse("")
+    // It looks like there are no articules with multiple duplicate PMCIDs, but let's be paranoid
```

Comment:
Is this a typo, or an awesome word for a tiny article?

Reply:
Hahaha, alas, it is a typo. Fixed in fba4082.